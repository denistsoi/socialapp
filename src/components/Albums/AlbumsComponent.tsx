import React, { ReactElement } from "react";
import { View, Text, StyleSheet, Dimensions } from "react-native";

const deviceWidth = Dimensions.get("window").width;

type Props = {
  albums: Array<Album> | null;
  onPress: Function;
};

import { Album as AlbumType, Album } from "../../types/resource";
import { TouchableOpacity } from "react-native-gesture-handler";

type AlbumPropType = {
  album: AlbumType;
  onPress: Function;
};

const styles = StyleSheet.create({
  album: {
    width: deviceWidth,
    flex: 1
  },
  imagePlaceholder: {
    height: 40,
    width: 40,
    backgroundColor: "grey",
    opacity: 0.7
  },
  touchable: {
    alignContent: "center",
    flex: 1
  }
});

const AlbumComponent = ({ album, onPress }: AlbumPropType): ReactElement => {
  const { id, title } = album;

  return (
    <View style={styles.album}>
      <TouchableOpacity
        onPress={(): void => onPress(id)}
        style={styles.touchable}>
        <View style={styles.imagePlaceholder}></View>
        <Text>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

const UserAlbums = ({ albums, onPress }: Props): ReactElement => {
  return (
    <View>
      {albums?.map((album, index) => (
        <AlbumComponent
          key={index}
          album={album}
          onPress={onPress}></AlbumComponent>
      ))}
    </View>
  );
};

export default UserAlbums;
