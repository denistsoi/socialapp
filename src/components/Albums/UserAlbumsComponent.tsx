import React, { ReactElement } from "react";
import { View, Text } from "react-native";

type Props = {
  albums: Array<Album> | null;
  onPress: Function;
};

import { Album as AlbumType, Album } from "../../types/resource";
import { TouchableOpacity } from "react-native-gesture-handler";

type AlbumPropType = {
  album: AlbumType;
  onPress: Function;
};
const AlbumComponent = ({ album, onPress }: AlbumPropType): ReactElement => {
  const { id, title } = album;

  return (
    <View>
      <TouchableOpacity onPress={(): void => onPress(id)}>
        <Text>
          {id} {title}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

const UserAlbums = ({ albums, onPress }: Props): ReactElement => {
  return (
    <View>
      {albums?.map((album, index) => (
        <AlbumComponent
          key={index}
          album={album}
          onPress={onPress}></AlbumComponent>
        // <View>
        //   <Text>
        //     {album.id} {album.title}
        //   </Text>
        // </View>
      ))}
    </View>
  );
};

export default UserAlbums;
