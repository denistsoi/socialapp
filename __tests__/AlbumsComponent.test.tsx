import React from "react";
import { render } from "react-native-testing-library";
import Albums from "../src/components/Albums/AlbumsComponent";

describe("Albums Component", () => {
  it.only("should render data in albums", async () => {
    let props: any;

    props = {
      onPress: jest.fn(),
      albums: [
        {
          id: 1,
          userId: 1,
          title: "Bob and the Bobsleigh"
        }
      ]
    };

    const component = render(
      <Albums onPress={props.onPress} albums={props.albums} />
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText("Bob and the Bobsleigh")).toBeDefined();
  });
});
