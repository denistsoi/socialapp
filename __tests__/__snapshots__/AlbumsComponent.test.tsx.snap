// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums Component should render data in albums 1`] = `
<View
  style={Object {}}
>
  <RCTScrollView
    collapsable={false}
    horizontal={true}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fafafa",
                "borderColor": "#888",
                "borderWidth": 0.5,
                "height": 60,
                "marginRight": 20,
                "padding": 2,
                "width": 60,
              }
            }
          >
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#666",
                  "fontSize": 10,
                }
              }
            >
              Bob and the Bobsleigh
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </RCTScrollView>
</View>
`;
